lib/csdl2openapi.js

```diff
@@ -1483,7 +1483,8 @@ module.exports.csdl2openapi = function (
         const updName = allName + '-update';
         const isKey = keyMap(type);
         const required = Object.keys(isKey);
-        const allProperties = {};
+        const metadata = { '@odata.id':{ type: "string", readOnly: true }, '@odata.etag': { type: 'string', readOnly: true, example: "W/\"08D7D40891852C76\"" } };

```

Comment:
I didn't correct the tests for this as I wanted to double check I was on right path before changing them :)